src/app/shared/services/task/task.service.ts

```diff
@@ -3,34 +3,50 @@ import { TemplateFieldService } from "../../components/template/services/templat
 import { AppDataService } from "../data/app-data.service";
 import { arrayToHashmap } from "../../utils";
 import { AsyncServiceBase } from "../asyncService.base";
+import { AppConfigService } from "../app-config/app-config.service";
+import { IAppConfig } from "../../model";
 
 @Injectable({
   providedIn: "root",
 })
 export class TaskService extends AsyncServiceBase {
```

Comment:
nit(blocking)
It would be useful to have a few basic tests for public methods. I appreciate there's quite a lot exposed via config, but at least having some basic mock data should help understand the main ideas (as it stands very hard to evaluate or review the logic for setting/getting tasks)